src/AElf.OS.Network.Grpc/GrpcPeer.cs

```diff
@@ -250,17 +250,26 @@ private void HandleFailure(AggregateException exceptions, string errorMessage)
                     // Either we connected again or the state change wait timed out.
                     if (_channel.State == ChannelState.TransientFailure || _channel.State == ChannelState.Connecting)
                     {
-                        await StopAsync();
-                        DisconnectionEvent?.Invoke(this, EventArgs.Empty);
+                        await StopAndRemove();
                     }
                 });
             }
+            else if(exceptions.InnerException is RpcException rpcEx && rpcEx.StatusCode == StatusCode.Cancelled)
+            {
+                _ = StopAndRemove();
+            }
             else
             {
                 throw new NetworkException($"Failed request to {this}: {errorMessage}", exceptions);
             }
         }
 
+        private async Task StopAndRemove()
+        {
+            await StopAsync();
+            DisconnectionEvent?.Invoke(this, EventArgs.Empty);
```

Comment:
what's this ?

Reply:
This method is used to shutdown the peer, the DisconnectionEvent is subscribed by the peer pool, when this is fired the peer pool will remove this peer from its internal list of peers.
So the event is fired in 2 cases for now:
1. The situation that this PR fixes (see description).
2. When the service uses a peer to request or broadcast and that the underlying grpc channel is not healthy (TRANSIENT_FAILURE, it can happen, maybe the channel is bad for just 1 or 2 seconds), the logic will wait a short time for the channel to recover, if after the wait it is still not recovered it will call StopAndRemoveAsync.

Reply:
we should not use C# event

just use EventBus

Reply:
ok, updated